### PR TITLE
feat(sgx): SGX_IOC_ENCLAVE_{RELAX,RESTRICT}_PERMISSIONS

### PR DESCRIPTION
**Pull request by jarkkojs**

### CODE REVIEW

**Comment by jarkkojs** [comment]
Consider this first PR for SGX2 as a probe how infrastructure etc. works.

File: src/backend/sgx/ioctls.rs [hunk 1]
```diff
@@ -102,3 +108,58 @@ impl<'a> SetAttribute<'a> {
...
+/// SGX_IOC_ENCLAVE_RELAX_PERMISSIONS parameter structure
+pub struct RelaxPermissions<'a> {
+    /// In: starting page offset
+    offset: u64,
+    /// In: length of the address range (multiple of the page size)
+    length: u64,
+    /// In: SECINFO containing the relaxed permissions
+    secinfo: u64,
+    /// Out: length of the address range successfully changed
+    count: u64,
```
Comment (npmccallum) [inline src/backend/sgx/ioctls.rs:123]: @jarkkojs If this ioctl does both input and output, shouldn't it be `_IOWR()` and therefore `WriteRead`?

File: src/backend/sgx/ioctls.rs [hunk 2]
```diff
@@ -102,3 +108,58 @@ impl<'a> SetAttribute<'a> {
...
+    /// Create a new RelaxPermissions instance.
+    pub fn new(offset: usize, length: usize, secinfo: &'a SecInfo) -> Self {
+        Self {
+            offset: offset as _,
+            length: length as _,
+            secinfo: secinfo as *const _ as _,
+            count: 0,
+            phantom: PhantomData,
+        }
+    }
```
Comment (npmccallum) [inline src/backend/sgx/ioctls.rs:137]: I think we need a `fn count(&self) -> usize` here.

File: src/backend/sgx/ioctls.rs (hunk 1 above)
Comment (jarkkojs) [reply src/backend/sgx/ioctls.rs:123]: Yes, it is `_IOWR()`, you're right.

Comment (jarkkojs) [reply src/backend/sgx/ioctls.rs:123]: For reference https://lore.kernel.org/linux-sgx/f0029cda125be9eb4eb2e051ab41d500081c6e2c.1644274683.git.reinette.chatre@intel.com/

**Comment by npmccallum** [comment]
@jarkkojs Please rebase.

**Comment by npmccallum** [comment]
@jarkkojs Looks like you need `feat(sgx)` rather than `feat[sgx]` to pass the tests.

**Review by npmccallum — DISMISSED** [review]

**Comment by jarkkojs** [comment]
I fixed the conventional commit message issues and did a new push.

**Review by npmccallum — APPROVED** [review]

**Pull request merged**